libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -172,13 +177,31 @@ class MapboxRouteLineApi(
      * @return a state which contains the side effects to be applied to the map displaying the
      * newly designated route line.
      */
-    fun updateToPrimaryRoute(route: DirectionsRoute): Expected<RouteSetValue, RouteLineError> {
-        val newRoutes = directionsRoutes.filter { it != route }.toMutableList().also {
-            it.add(0, route)
+    suspend fun updateToPrimaryRoute(route: DirectionsRoute):
+        Expected<RouteSetValue, RouteLineError> {
+            val newRoutes = directionsRoutes.filter { it != route }.toMutableList().also {
+                it.add(0, route)
+            }
+            val featureDataProvider: () -> List<RouteFeatureData> =
+                MapboxRouteLineUtils.getRouteFeatureDataProvider(newRoutes)
+            return setNewRouteData(newRoutes, featureDataProvider)
+        }
+
+    /**
+     * Updates which route is identified as the primary route.
+     *
+     * @param route the [DirectionsRoute] which should be designated as the primary
+     * @param consumer a consumer of the result of calling this method which should be applied to
+     * a render method in [MapboxRouteLineView]
+     */
+    fun updateToPrimaryRoute(
+        route: DirectionsRoute,
+        consumer: MapboxNavigationConsumer<Expected<RouteSetValue, RouteLineError>>
+    ) {
+        ThreadController.getMainScopeAndRootJob().scope.launch {
```

Comment:
This uses a different job than `setRoutes` so we can run into race conditions where calling `setRoutes` would not actually apply the changes because the result would be overwritten by `updateToPrimaryRoute`.